### ISSUE TITLE
`dotnet workload install` and maui-check fail [Bug] 

### ISSUE THREAD

**Issue by Ultra-max99** [dotnet/maui#2159]
### Description

Updating preview 6 to preview 7 fails with errors below.

### Steps to Reproduce

Have MAUI workloads for preview 6 installed.

Run `maui-check` or `dotnet workload install`.

`maui-check` output:

```
      _   _   _____   _____     __  __      _      _   _   ___
     | \ | | | ____| |_   _|   |  \/  |    / \    | | | | |_ _|
     |  \| | |  _|     | |     | |\/| |   / _ \   | | | |  | |
  _  | |\  | | |___    | |     | |  | |  / ___ \  | |_| |  | |
 (_) |_| \_| |_____|   |_|     |_|  |_| /_/   \_\  \___/  |___|

☼ .NET MAUI Check v0.6.1.0 ☼
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
This tool will attempt to evaluate your .NET MAUI development environment.
If problems are detected, this tool may offer the option to try and fix them for you, or suggest a way to fix them
yourself.

Thanks for choosing .NET MAUI!
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
» Synchronizing configuration...
Error: Error resolving type specified in JSON 'DotNetCheck.Manifest.NuGetFeedVariableMapper, MauiCheck'. Path
'check.variableMappers[0].$type', line 9, position 74.
```

`dotnet workload install maui` output:

```

Welcome to .NET 6.0!
---------------------
SDK Version: 6.0.100-preview.7.21379.14

Telemetry
---------
The .NET tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only).
Learn about HTTPS: https://aka.ms/dotnet-https
----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------

Updated advertising manifest microsoft.net.sdk.android.
Updated advertising manifest microsoft.net.workload.mono.toolchain.
Updated advertising manifest microsoft.net.sdk.maui.
Updated advertising manifest microsoft.net.sdk.ios.
Updated advertising manifest microsoft.net.sdk.macos.
Updated advertising manifest microsoft.net.sdk.maccatalyst.
Updated advertising manifest microsoft.net.workload.emscripten.
Updated advertising manifest microsoft.net.sdk.tvos.
Installing workload manifest microsoft.net.sdk.android version 30.0.100-preview.7.110.
Installing workload manifest microsoft.net.sdk.ios version 15.0.100-preview.7231.
Installing workload manifest microsoft.net.sdk.maccatalyst version 15.0.100-preview.7231.
Installing workload manifest microsoft.net.sdk.macos version 12.0.100-preview.7231.
Installing workload manifest microsoft.net.sdk.maui version 6.0.100-preview.7.1345.
Installing workload manifest microsoft.net.sdk.tvos version 15.0.100-preview.7231.
Installing pack Microsoft.Maui.Core.Ref.android version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Ref.android version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Core.Runtime.android version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Runtime.android version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Ref.android version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Ref.android version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Runtime.android version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Runtime.android version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Ref.android version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Ref.android version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Runtime.android version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Runtime.android version 6.0.100-preview.7.1345...
Installing pack Microsoft.AspNetCore.Components.WebView.Maui version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.AspNetCore.Components.WebView.Maui version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Dependencies version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Dependencies version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Sdk version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Sdk version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Extensions version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Extensions version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Resizetizer.Sdk version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Resizetizer.Sdk version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Templates version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Templates version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Core.Ref.any version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Ref.any version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Core.Runtime.any version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Runtime.any version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Ref.any version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Ref.any version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Runtime.any version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Runtime.any version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Ref.any version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Ref.any version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Runtime.any version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Runtime.any version 6.0.100-preview.7.1345...
Installing pack Microsoft.Android.Sdk version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Sdk version 30.0.100-preview.7.110...
Installing pack Microsoft.Android.Sdk.BundleTool version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Sdk.BundleTool version 30.0.100-preview.7.110...
Installing pack Microsoft.Android.Ref version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Ref version 30.0.100-preview.7.110...
Installing pack Microsoft.Android.Runtime.android-arm version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Runtime.android-arm version 30.0.100-preview.7.110...
Installing pack Microsoft.Android.Runtime.android-arm64 version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Runtime.android-arm64 version 30.0.100-preview.7.110...
Installing pack Microsoft.Android.Runtime.android-x86 version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Runtime.android-x86 version 30.0.100-preview.7.110...
Installing pack Microsoft.Android.Runtime.android-x64 version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Runtime.android-x64 version 30.0.100-preview.7.110...
Installing pack Microsoft.Android.Templates version 30.0.100-preview.7.110...
Writing workload pack installation record for Microsoft.Android.Templates version 30.0.100-preview.7.110...
Installing pack Microsoft.NETCore.App.Runtime.Mono.android-arm version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.android-arm version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.android-arm64 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.android-arm64 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.android-x64 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.android-x64 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.android-x86 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.android-x86 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NET.Runtime.MonoAOTCompiler.Task version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NET.Runtime.MonoAOTCompiler.Task version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NET.Runtime.MonoTargets.Sdk version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NET.Runtime.MonoTargets.Sdk version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.Maui.Core.Ref.ios version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Ref.ios version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Core.Runtime.ios version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Runtime.ios version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Ref.ios version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Ref.ios version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Runtime.ios version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Runtime.ios version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Ref.ios version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Ref.ios version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Runtime.ios version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Runtime.ios version 6.0.100-preview.7.1345...
Installing pack Microsoft.iOS.Sdk version 15.0.100-preview.7.230...
Writing workload pack installation record for Microsoft.iOS.Sdk version 15.0.100-preview.7.230...
Installing pack Microsoft.iOS.Windows.Sdk version 15.0.100-preview.7.230...
Writing workload pack installation record for Microsoft.iOS.Windows.Sdk version 15.0.100-preview.7.230...
Installing pack Microsoft.iOS.Ref version 15.0.100-preview.7.230...
Writing workload pack installation record for Microsoft.iOS.Ref version 15.0.100-preview.7.230...
Installing pack Microsoft.iOS.Templates version 15.0.100-preview.7.230...
Writing workload pack installation record for Microsoft.iOS.Templates version 15.0.100-preview.7.230...
Installing pack Microsoft.NETCore.App.Runtime.Mono.ios-arm version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.ios-arm version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.ios-arm64 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.ios-arm64 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.iossimulator-arm64 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.iossimulator-arm64 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.iossimulator-x64 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.iossimulator-x64 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.iossimulator-x86 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.iossimulator-x86 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.Maui.Core.Ref.maccatalyst version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Ref.maccatalyst version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Core.Runtime.maccatalyst version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Runtime.maccatalyst version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Ref.maccatalyst version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Ref.maccatalyst version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Runtime.maccatalyst version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Runtime.maccatalyst version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Ref.maccatalyst version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Ref.maccatalyst version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Runtime.maccatalyst version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Runtime.maccatalyst version 6.0.100-preview.7.1345...
Installing pack Microsoft.MacCatalyst.Sdk version 15.0.100-preview.7.230...
Writing workload pack installation record for Microsoft.MacCatalyst.Sdk version 15.0.100-preview.7.230...
Installing pack Microsoft.MacCatalyst.Ref version 15.0.100-preview.7.230...
Writing workload pack installation record for Microsoft.MacCatalyst.Ref version 15.0.100-preview.7.230...
Installing pack Microsoft.MacCatalyst.Templates version 15.0.100-preview.7.230...
Writing workload pack installation record for Microsoft.MacCatalyst.Templates version 15.0.100-preview.7.230...
Installing pack Microsoft.NETCore.App.Runtime.Mono.maccatalyst-arm64 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.maccatalyst-arm64 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.NETCore.App.Runtime.Mono.maccatalyst-x64 version 6.0.0-preview.7.21377.19...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.maccatalyst-x64 version 6.0.0-preview.7.21377.19...
Installing pack Microsoft.Maui.Core.Ref.win version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Ref.win version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Core.Runtime.win version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Core.Runtime.win version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Ref.win version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Ref.win version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Controls.Runtime.win version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Controls.Runtime.win version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Ref.win version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Ref.win version 6.0.100-preview.7.1345...
Installing pack Microsoft.Maui.Essentials.Runtime.win version 6.0.100-preview.7.1345...
Writing workload pack installation record for Microsoft.Maui.Essentials.Runtime.win version 6.0.100-preview.7.1345...
Garbage collecting for SDK feature bands 6.0.100...
Workload installation failed: The JSON value could not be converted to Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadPackId. Path: $.Id | LineNumber: 0 | BytePositionInLine: 29.
install
  Install a workload.

Usage:
  dotnet [options] workload install [<WORKLOAD_ID>...]

Arguments:
  <WORKLOAD_ID>  The NuGet Package Id of the workload to install.

Options:
  --sdk-version <VERSION>                                  The version of the SDK.
  --configfile <FILE>                                      The NuGet configuration file to use.
  -s, --source <SOURCE>                                    The NuGet package source to use during the restore.
  --skip-manifest-update                                   Skip updating the workload manifests.
  --from-cache <from-cache>                                Complete the operation from cache (offline).
  --download-to-cache <download-to-cache>                  Download packages needed to install a workload to a folder
                                                           which can be used for offline installation.
  --include-previews                                       Allow prerelease workload manifests.
  --temp-dir <temp-dir>                                    Configure the temporary directory used for this command
                                                           (must be secure).
  --disable-parallel                                       Prevent restoring multiple projects in parallel.
  --ignore-failed-sources                                  Treat package source failures as warnings.
  --no-cache                                               Do not cache packages and http requests.
  --interactive                                            Allows the command to stop and wait for user input or action 
                                                           (for example to complete authentication).
  -v, --verbosity                                          Set the MSBuild verbosity level. Allowed values are q[uiet], 
  <d|detailed|diag|diagnostic|m|minimal|n|normal|q|quiet>  m[inimal], n[ormal], d[etailed], and diag[nostic].
  -?, -h, --help                                           Show help and usage information


```

`dotnet workload uninstall maui` output:
```

Removing workload installation record for maui...
Garbage collecting for SDK feature bands 6.0.100...
Workload uninstallation failed: The JSON value could not be converted to Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadPackId. Path: $.Id | LineNumber: 0 | BytePositionInLine: 29.
uninstall
  Uninstall a workload.

Usage:
  dotnet [options] workload uninstall [<WORKLOAD_ID>...]

Arguments:
  <WORKLOAD_ID>  The NuGet Package Id of the workload to install.

Options:
  -?, -h, --help  Show help and usage information


```
### Expected Behavior

No Errors.

### Actual Behavior

Errors as above.

### Basic Information

- Version with issue: preview 6 to 7.
- Last known good version: preview 6.

**Comment by dhusemann** [comment]
stuck also 
❯ dotnet --version
6.0.100-preview.7.21379.14
   drhen@LAPTOP-K1ES7NNB  ~ 
❯ dotnet workload install maui
Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestCompositionException: Workload pack 'Microsoft.NET.Runtime.Emscripten.Node' in manifest 'microsoft.net.workload.mono.toolchain' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.workload.mono.toolchain\WorkloadManifest.json] conflicts with manifest 'microsoft.net.workload.emscripten' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.workload.emscripten\WorkloadManifest.json]
   at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.ComposeWorkloadManifests() in Microsoft.DotNet.TemplateLocator.dll:token 0x6000091+0x2be
   at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.Create(IWorkloadManifestProvider manifestProvider, String dotnetRootPath, String sdkVersion) in Microsoft.DotNet.TemplateLocator.dll:token 0x600008b+0x8a
   at Microsoft.DotNet.Workloads.Workload.Install.WorkloadInstallCommand..ctor(ParseResult parseResult, IReporter reporter, IWorkloadResolver workloadResolver, IInstaller workloadInstaller, INuGetPackageDownloader nugetPackageDownloader, IWorkloadManifestUpdater workloadManifestUpdater, String dotnetDir, String userHome, String tempDirPath, String version, IReadOnlyCollection`1 workloadIds) in dotnet.dll:token 0x60002d2+0x1d1
   at Microsoft.DotNet.Workloads.Workload.WorkloadCommand.<get_SubCommands>b__9_0(ParseResult appliedOption) in dotnet.dll:token 0x6000204+0x0
   at Microsoft.DotNet.Cli.DotNetTopLevelCommandBase.RunCommand(String[] args) in dotnet.dll:token 0x6000a56+0x44
   at Microsoft.DotNet.Workloads.Workload.WorkloadCommand.Run(String[] args) in dotnet.dll:token 0x6000202+0x6
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, TimeSpan startupTime, ITelemetry telemetryClient) in dotnet.dll:token 0x6000ac1+0x2e8
   at Microsoft.DotNet.Cli.Program.Main(String[] args) in dotnet.dll:token 0x6000abf+0x6f
   drhen@LAPTOP-K1ES7NNB  ~  ERROR 
❯ dotnet tool install -g redth.net.maui.check
C:\Users\drhen\AppData\Local\Temp\2u53r4cu.qsr\restore.csproj(5,3): warning MSB4242: The SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" failed to run. Workload pack 'Microsoft.NET.Runtime.Emscripten.Node' in manifest 'microsoft.net.workload.mono.toolchain' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.workload.mono.toolchain\WorkloadManifest.json] conflicts with manifest 'microsoft.net.workload.emscripten' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.workload.emscripten\WorkloadManifest.json]
C:\Program Files\dotnet\sdk\6.0.100-preview.7.21379.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props(14,3): warning MSB4242: The SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" failed to run. Workload pack 'Microsoft.NET.Runtime.Emscripten.Node' in manifest 'microsoft.net.workload.mono.toolchain' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.workload.mono.toolchain\WorkloadManifest.json] conflicts with manifest 'microsoft.net.workload.emscripten' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.workload.emscripten\WorkloadManifest.json]
C:\Program Files\dotnet\sdk\6.0.100-preview.7.21379.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props(14,38): error MSB4236: The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found. [C:\Users\drhen\AppData\Local\Temp\2u53r4cu.qsr\restore.csproj]
The tool package could not be restored.
Tool 'redth.net.maui.check' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

**Comment by dhusemann** [comment]
can't load maui-check for the same reason I can't run the workload, got stuck in the initial update to preview 3 of 2022,  have completedly unloaded, 2019 and 2022 and just loaded 2022 with dotnet 6 preview 7 with the same results.

**Comment by pierre01** [comment]
I have the same issue PS

 C:\Users\pierr> dotnet workload install maui

Updated advertising manifest microsoft.net.workload.emscripten.
Updated advertising manifest microsoft.net.sdk.android.
Updated advertising manifest microsoft.net.sdk.maccatalyst.
Updated advertising manifest microsoft.net.sdk.maui.
Updated advertising manifest microsoft.net.workload.mono.toolchain.
Updated advertising manifest microsoft.net.sdk.macos.
Updated advertising manifest microsoft.net.sdk.ios.
Updated advertising manifest microsoft.net.sdk.tvos.
Installing workload manifest microsoft.net.sdk.android version 31.0.100-ci.d16-11.6.
Installing workload manifest microsoft.net.sdk.ios version 15.0.100-rc.1.523.
Installing workload manifest microsoft.net.sdk.maccatalyst version 15.0.100-rc.1.523.
Installing workload manifest microsoft.net.sdk.macos version 12.0.100-rc.1.523.
Installing workload manifest microsoft.net.sdk.maui version 6.0.100-rc.2.1594.
Installing workload manifest microsoft.net.sdk.tvos version 15.0.100-rc.1.523.
Installing pack Microsoft.Maui.Core.Ref.android version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Core.Ref.android version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Core.Runtime.android version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Core.Runtime.android version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Controls.Ref.android version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Controls.Ref.android version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Controls.Runtime.android version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Controls.Runtime.android version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Essentials.Ref.android version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Essentials.Ref.android version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Essentials.Runtime.android version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Essentials.Runtime.android version 6.0.100-rc.2.1594...
Installing pack Microsoft.AspNetCore.Components.WebView.Maui version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.AspNetCore.Components.WebView.Maui version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Dependencies version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Dependencies version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Controls.Sdk version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Controls.Sdk version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Extensions version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Extensions version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Resizetizer.Sdk version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Resizetizer.Sdk version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Templates version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Templates version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Core.Ref.any version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Core.Ref.any version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Core.Runtime.any version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Core.Runtime.any version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Controls.Ref.any version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Controls.Ref.any version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Controls.Runtime.any version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Controls.Runtime.any version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Essentials.Ref.any version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Essentials.Ref.any version 6.0.100-rc.2.1594...
Installing pack Microsoft.Maui.Essentials.Runtime.any version 6.0.100-rc.2.1594...
Writing workload pack installation record for Microsoft.Maui.Essentials.Runtime.any version 6.0.100-rc.2.1594...
Workload installation failed, rolling back installed packs...
Installing workload manifest microsoft.net.sdk.android version 30.0.100-preview.7.110.
Installing workload manifest microsoft.net.sdk.ios version 15.0.100-preview.7.230.
Installing workload manifest microsoft.net.sdk.maccatalyst version 15.0.100-preview.7.230.
Installing workload manifest microsoft.net.sdk.macos version 12.0.100-preview.7.230.
Installing workload manifest microsoft.net.sdk.maui version 6.0.100-preview.7.1345.
Installing workload manifest microsoft.net.sdk.tvos version 15.0.100-preview.7.230.
Installation roll back failed: Could not find workload 'android' extended by workload 'maui-android' in manifest 'microsoft.net.sdk.maui' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.sdk.maui\WorkloadManifest.json]
Rolling back pack Microsoft.Maui.Core.Ref.android installation...
Uninstalling workload pack Microsoft.Maui.Core.Ref.android version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Core.Runtime.android installation...
Uninstalling workload pack Microsoft.Maui.Core.Runtime.android version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Controls.Ref.android installation...
Uninstalling workload pack Microsoft.Maui.Controls.Ref.android version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Controls.Runtime.android installation...
Uninstalling workload pack Microsoft.Maui.Controls.Runtime.android version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Essentials.Ref.android installation...
Uninstalling workload pack Microsoft.Maui.Essentials.Ref.android version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Essentials.Runtime.android installation...
Uninstalling workload pack Microsoft.Maui.Essentials.Runtime.android version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.AspNetCore.Components.WebView.Maui installation...
Uninstalling workload pack Microsoft.AspNetCore.Components.WebView.Maui version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Dependencies installation...
Uninstalling workload pack Microsoft.Maui.Dependencies version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Controls.Sdk installation...
Uninstalling workload pack Microsoft.Maui.Controls.Sdk version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Extensions installation...
Uninstalling workload pack Microsoft.Maui.Extensions version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Resizetizer.Sdk installation...
Uninstalling workload pack Microsoft.Maui.Resizetizer.Sdk version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Templates installation...
Uninstalling workload pack Microsoft.Maui.Templates version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Core.Ref.any installation...
Uninstalling workload pack Microsoft.Maui.Core.Ref.any version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Core.Runtime.any installation...
Uninstalling workload pack Microsoft.Maui.Core.Runtime.any version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Controls.Ref.any installation...
Uninstalling workload pack Microsoft.Maui.Controls.Ref.any version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Controls.Runtime.any installation...
Uninstalling workload pack Microsoft.Maui.Controls.Runtime.any version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Essentials.Ref.any installation...
Uninstalling workload pack Microsoft.Maui.Essentials.Ref.any version 6.0.100-rc.2.1594.
Rolling back pack Microsoft.Maui.Essentials.Runtime.any installation...
Uninstalling workload pack Microsoft.Maui.Essentials.Runtime.any version 6.0.100-rc.2.1594.
Workload installation failed: Could not find workload 'android' extended by workload 'maui-android' in manifest 'microsoft.net.sdk.maui' [C:\Program Files\dotnet\sdk-manifests\6.0.100\microsoft.net.sdk.maui\WorkloadManifest.json]
install

**Comment by nurbat** [comment]
![изображение](https://user-images.githubusercontent.com/5789051/132320980-ee5be58a-eb6c-430e-8831-e42a6e496baa.png)

**Comment by jfversluis** [comment]
Hey everyone! Sorry you are having issues. Things are moving fast with the previews and because of that, these versions are already "old". You should not need to use `maui-check` anymore and can install this from the Visual Studio installer or with the `dotnet workload` tool. I'm going to assume you got it working in the meantime, if not, please open a new issue with all the details you have.

Thanks!
